service/src/main/java/com/softserve/borda/services/impl/BoardServiceImpl.java

```diff
@@ -70,4 +74,17 @@ public boolean deleteBoardById(Long id) {
                 .map(UserBoardRelation::getBoard)
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public Board addTagToBoard(Long boardId, Long tagId) {
+        Board board = getBoardById(boardId);
+        Tag tag = tagService.getTagById(tagId);
+        board.getTags().add(tag);
```

Comment:
We do not need to touch board to save a tag.
As I understand, Tag entity has boarId inside and you can save only tag without updating the whole board.

Reply:
Board has list of tags. Like labels from Trello. And we need them to always be connected to some board. Tag has no board inside right now.